### PR TITLE
openapi(engine): add filter and limit to list job

### PR DESCRIPTION
**Pull request by gozssky**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/7654

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
None
```

### CODE REVIEW

File: engine/servermaster/jobmanager.go [hunk 1]
```diff
@@ -224,13 +226,25 @@ func (jm *JobManagerImpl) GetJob(ctx context.Context, req *pb.GetJobRequest) (*p
...
-		setDetailToMasterMeta(masterMeta, detail, err)
+		if err != nil {
+			job.Error = &pb.Job_Error{
+				Code:    "", // TODO: extract error code from err.
+				Message: err.Error(),
+			}
+		} else {
+			job.Detail = detail
+		}
 	}
```
Comment (amyangfei) [inline engine/servermaster/jobmanager.go:245]: Can extract this logic into a function and reuse it at L412

Comment (gozssky) [reply engine/servermaster/jobmanager.go]: Addressed in latest commits.

File: engine/servermaster/jobmanager.go [hunk 2]
```diff
@@ -482,7 +486,7 @@ func (jm *JobManagerImpl) GetJobMasterForwardAddress(ctx context.Context, jobID
 		}
 		return "", err
 	}
-	if masterMeta.State != frameModel.MasterStateInit || jm.JobFsm.QueryOnlineJob(jobID) == nil {
+	if masterMeta.State != frameModel.MasterStateInit && jm.JobFsm.QueryOnlineJob(jobID) == nil {
```
Comment (CharlesCheung96) [inline engine/servermaster/jobmanager.go:489]: Why change the logic here?

Comment (gozssky) [reply engine/servermaster/jobmanager.go:489]: This's a bug. We shouldn't forward requests to the job master when the job is not online.
It's also the root cause of https://github.com/pingcap/tiflow/issues/7555.

Comment (amyangfei) [reply engine/servermaster/jobmanager.go:489]: Confused, does `job online` mean
`masterMeta.State == frameModel.MasterStateInit && jm.JobFsm.QueryOnlineJob(jobID) != nil`
Seems the original check is correct

Comment (gozssky) [inline engine/servermaster/jobmanager.go:489]:
```suggestion
	if masterMeta.State != frameModel.MasterStateInit || jm.JobFsm.QueryOnlineJob(jobID) == nil {
```

Comment (gozssky) [reply engine/servermaster/jobmanager.go]: Fixed.

**Review by amyangfei — APPROVED** [review]

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**